File: services/arona/misskey-mixi2-link/store/store_test.go
package store

import (
	"database/sql"
	"testing"

	_ "modernc.org/sqlite"
)

func newTestStore(t *testing.T) *Store {
	t.Helper()
	db, err := sql.Open("sqlite", ":memory:")
	if err != nil {
		t.Fatalf("open sqlite: %v", err)
	}
	t.Cleanup(func() { db.Close() })
	s, err := Open(db)
	if err != nil {
		t.Fatalf("open store: %v", err)
	}
	return s
}

func TestRecordThenIsProcessed(t *testing.T) {
	s := newTestStore(t)
	if processed, err := s.IsProcessed("misskey->mixi2", "n1"); err != nil || processed {
		t.Fatalf("expected not processed, got processed=%v err=%v", processed, err)
	}
	if err := s.Record("misskey->mixi2", "n1", "p1"); err != nil {
		t.Fatalf("record: %v", err)
	}
	if processed, err := s.IsProcessed("misskey->mixi2", "n1"); err != nil || !processed {
		t.Fatalf("expected processed, got processed=%v err=%v", processed, err)
	}
}

func TestDirectionIsPartOfIdentity(t *testing.T) {
	s := newTestStore(t)
	if err := s.Record("misskey->mixi2", "id1", "t1"); err != nil {
		t.Fatalf("record: %v", err)
	}
	if processed, err := s.IsProcessed("mixi2->misskey", "id1"); err != nil || processed {
		t.Fatalf("expected not processed for other direction, got processed=%v err=%v", processed, err)
	}
}

func TestDuplicateRecordDoesNotError(t *testing.T) {
	s := newTestStore(t)
	if err := s.Record("misskey->mixi2", "n1", "p1"); err != nil {
		t.Fatalf("first record: %v", err)
	}
	if err := s.Record("misskey->mixi2", "n1", "p2"); err != nil {
		t.Fatalf("duplicate record should not error: %v", err)
	}
}

func TestCursorSetAndOverwrite(t *testing.T) {
	s := newTestStore(t)
	if v, err := s.GetCursor("misskey_last_note_id"); err != nil || v != nil {
		t.Fatalf("expected nil cursor, got v=%v err=%v", v, err)
	}
	if err := s.SetCursor("misskey_last_note_id", "a"); err != nil {
		t.Fatalf("set cursor a: %v", err)
	}
	if err := s.SetCursor("misskey_last_note_id", "b"); err != nil {
		t.Fatalf("set cursor b: %v", err)
	}
	v, err := s.GetCursor("misskey_last_note_id")
	if err != nil {
		t.Fatalf("get cursor: %v", err)
	}
	if v == nil || *v != "b" {
		t.Fatalf("expected cursor \"b\", got %v", v)
	}
}

File: services/connections/misskey-mixi2-link/test/store.test.ts
import { describe, expect, it } from "bun:test";
import { BridgeStore } from "../src/store.js";

describe("BridgeStore", () => {
  it("記録した source_id は処理済みになる", async () => {
    const store = await BridgeStore.open(":memory:");
    expect(await store.isProcessed("misskey->mixi2", "n1")).toBe(false);
    await store.record("misskey->mixi2", "n1", "p1");
    expect(await store.isProcessed("misskey->mixi2", "n1")).toBe(true);
  });

  it("方向が違えば別の記録になる", async () => {
    const store = await BridgeStore.open(":memory:");
    await store.record("misskey->mixi2", "id1", "t1");
    expect(await store.isProcessed("mixi2->misskey", "id1")).toBe(false);
  });

  it("同じ source_id の二重記録はエラーにならない", async () => {
    const store = await BridgeStore.open(":memory:");
    await store.record("misskey->mixi2", "n1", "p1");
    expect(store.record("misskey->mixi2", "n1", "p2")).resolves.toBeUndefined();
  });

  it("カーソルの保存と上書き", async () => {
    const store = await BridgeStore.open(":memory:");
    expect(await store.getCursor("misskey_last_note_id")).toBeNull();
    await store.setCursor("misskey_last_note_id", "a");
    await store.setCursor("misskey_last_note_id", "b");
    expect(await store.getCursor("misskey_last_note_id")).toBe("b");
  });
});
